starport/services/scaffolder/component.go

```diff
@@ -207,3 +208,23 @@ func checkComponentCreated(appPath, moduleName string, compName multiformatname.
 	}
 	return err
 }
+
+// checkForbiddenOracleFieldName returns true if the name is forbidden as an oracle field name
+func checkForbiddenOracleFieldName(name string) error {
```

Comment:
This is a bit tricky but I may think the `name` parameter coming from `multiformatname.NewName` can contains hyphens or underscores since snake case or kebab case string can be passed to it

If it's the case it could be better to first get the `multiformatname` name and compare those value with a case that doesn't contain special character like lowerCamel